src/content/codelabs/dart-cheatsheet.md

```diff
@@ -1778,106 +1777,131 @@ class IntegerHolder {
+      2 => ('IntegerHolder.fromList([1, 2])', IntegerHolder.fromList([1, 2]), IntegerDouble),
+      3 => ('IntegerHolder.fromList([1, 2, 3])', IntegerHolder.fromList([1, 2, 3]), IntegerTriple),
+      _ => (null, null, null), // Default case if count doesn't match 1, 2, or 3
+    };
+    if (obj.runtimeType != expected) {
+      errs.add(
+          'Called $callTxt and got an object of type \n ${obj.runtimeType} instead of $expected.');
     } else {
-      if (obj.a != 1) {
-        errs.add('Called IntegerHolder.fromList([1]) and got an IntegerSingle with \n  an \'a\' value of ${obj.a} instead of the expected (1).');
+      if (obj is IntegerSingle) {
+        testValueOfA(errs,[1],obj,callTxt);
+      } else if (obj is IntegerDouble) {
+        testValueOfAB(errs,[1,2],obj,callTxt);
+      } else if (obj is IntegerTriple) {
+        testValueOfABC(errs,[1,2,3],obj,callTxt);
+      } else {
+        // Can't happen
       }
     }
   } catch (e) {
-    print('Called IntegerHolder.fromList([]) and got an exception of \n type ${e.runtimeType}.');
-    return;
+    print(
+        'Called ${callTxt} and got an exception of \n type ${e.runtimeType}.');
+    return false;
   }
+  return true;
+}
 
-  try {
-    final obj = IntegerHolder.fromList([1, 2]);
-    
-    if (obj is! IntegerDouble) {
-      errs.add('Called IntegerHolder.fromList([1, 2]) and got an object of type \n ${obj.runtimeType} instead of IntegerDouble.');
-    } else {
-      if (obj.a != 1) {
-        errs.add('Called IntegerHolder.fromList([1, 2]) and got an IntegerDouble \n with an \'a\' value of ${obj.a} instead of the expected (1).');
-      }
-      
-      if (obj.b != 2) {
-        errs.add('Called IntegerHolder.fromList([1, 2]) and got an IntegerDouble \n with an \'b\' value of ${obj.b} instead of the expected (2).');
-      }
-    }
-  } catch (e) {
-    print('Called IntegerHolder.fromList([1, 2]) and got an exception \n of type ${e.runtimeType}.');
-    return;
+testValueOfA(
+    List<String> errs, List<int> ex, IntegerSingle o, String callTxt) {
+  testValueOfX(errs, o.a, ex[0], 'a', callTxt, IntegerSingle);
+}
+
+testValueOfAB(List<String> errs, List<int> ex, IntegerDouble o, String callTxt) {
+  testValueOfX(errs, o.a, ex[0], 'a', callTxt, IntegerDouble);
+  testValueOfX(errs, o.b, ex[1], 'b', callTxt, IntegerDouble);
+}
+
+testValueOfABC(
+    List<String> errs, List<int> ex, IntegerTriple o, String callTxt) {
+  testValueOfX(errs, o.a, ex[0], 'a', callTxt, IntegerTriple);
+  testValueOfX(errs, o.b, ex[1], 'b', callTxt, IntegerTriple);
+  testValueOfX(errs, o.c, ex[2], 'c', callTxt, IntegerTriple);
+}
+
+testValueOfX(List<String> errs, int found, int ex, String name,
+    String callTxt, Type type) {
+  if (ex != found) {
+    errs.add(
+        'Called $callTxt and got a $type with \n a \'$name\' value of $found instead of the expected ($ex).');
   }
+}
 
+bool testBadNumberOfArgs(List<String> errs, int count) {
+  assert(count == 0 || count == 4);
```

Comment:
should this check be in a separate function that is to test the value as opposed to the test itself?